documentation/puppet_server_metrics.markdown

```diff
@@ -0,0 +1,423 @@
+>
+> -   The module disables the `iptables` firewall and enables cross-origin resource sharing on Apache, which are potential security risks.
+
+#### Installing the `grafanadash` Puppet module
+
+Install the `grafanadash` Puppet module on a \*nix agent. The module's `grafanadash::dev` class installs and configures a Graphite server, the Grafana extension, and a default dashboard.
+
+1.  [Install a \*nix Puppet agent](http://docs.puppet.com/puppet/latest/install_linux.html) to serve as the Graphite server.
+
+2.  As root on the Puppet agent node, run `puppet module install cprice404-grafanadash`.
+
+3.  As root on the Puppet agent node, run `puppet apply -e 'include grafanadash::dev'`.
+
+#### Running Grafana
+
+Grafana runs as a web dashboard, and the `grafanadash` module configures it to use port 10000 by default. To view Puppet metrics in Grafana, you must create a metrics dashboard, or edit and import a JSON-based dashboard that includes Puppet metrics, such as the [sample Grafana dashboard][] that we provide.
+
+1.  In a web browser on a computer that can reach the Puppet agent node running Grafana, navigate to `http://<AGENT'S HOSTNAME>:10000`.
+
+    There, you'll see a test screen that indicates whether Grafana can successfully connect to your Graphite server.
+
+    If Grafana is configured to use a hostname that the computer on which the browser is running cannot resolve, click **view details** and then the **Requests** tab to determine the hostname Grafana is trying to use. Next, add the IP address and hostname to the computer's `/etc/hosts` file on Linux or OS X, or `C:\Windows\system32\drivers\etc\hosts` file on Windows.
+
+2.  Download and edit our [sample Grafana dashboard][], `sample_metrics_dashboard.json`.
+
+    a.  Open the `sample_metrics_dashboard.json` file in a text editor on the same computer you're using to access Grafana.
+
+    b.  Throughout the file, replace our sample hostname of `master.example.com` with your Puppet Server's hostname. (**Note:** This value **must** be used as the `metrics_server_id` setting, as configured below.)
+
+    c.  Save the file.
+
+3.  In the Grafana UI, click **search** (the folder icon), then **Import**, then **Browse**.
+
+4.  Navigate to and select the edited JSON file.
+
+This loads a dashboard with nine graphs that display various metrics exported from the Puppet Server to the Graphite server. (For details, see [Using the Grafana dashboard](#using-the-sample-grafana-dashboard).) However, these graphs will remain empty until you enable Puppet Server's Graphite metrics.
+
+### Enabling Puppet Server's Graphite support
+
+Configure Puppet Server's [`metrics.conf`](./config_file_metrics.markdown) file to enable and use the Graphite server.
+
+1.  Set the `enabled` parameter to true in `metrics:registries:puppetserver:reporters:graphite`:
```

Comment:
Crap.... I was using dot notation to talk about drilling into hocon tables. Did I do that wrong? (see: [`metrics.metrics-webservice.jolokia.servlet-init-params`](https://github.com/puppetlabs/puppetserver/blob/master/documentation/metrics-api/v2/metrics_api.markdown#configuring-jolokia))